R/raincloudPlots.R

```diff
@@ -0,0 +1,908 @@
+    lwd = options$boxOutlineWidth, show.legend = FALSE
+  )
+
+  return(list(lowerWhiskers = lowerWhiskers, upperWhiskers = upperWhiskers))
+}  # End .rainWhiskers()
+
+
+
+# .rainMeansAndLines() ----
+.rainMeansAndLines <- function(options, boxPosVec, aesX, aesFill, infoFactorCombinations, meanPosition) {
+
+  means <- if (options$mean) {
+    ggplot2::stat_summary(
+      fun = mean, geom = "point",
+      mapping = ggplot2::aes(x = aesX, fill = aesFill),  # No color argument, covariate will interfere with it
+      color = .rainOutlineColor(options, "colorPalette", infoFactorCombinations),  # Instead like Outlines
+      shape = 18, size = options$meanSize, alpha = 1, show.legend = FALSE, position = meanPosition
+    )
+  } else {
+    NULL
+  }
+  meanLinesGroupMapping <- if (options$secondaryFactor == "") 1 else aesFill
+  meanLinesColor <- if (options$secondaryFactor == "") "black" else .rainOutlineColor(options, "colorPalette", infoFactorCombinations)
+  meanLines <- if (options$mean && options$meanLines) {  # Needs options$mean as qml wont uncheck options$meanLines
+    ggplot2::stat_summary(                                # if options$mean is unchecked again
+      fun = mean, geom = "line", mapping = ggplot2::aes(group = meanLinesGroupMapping),
+      color = meanLinesColor, alpha = options$meanLinesOpacity, position = meanPosition, lwd = options$meanLinesWidth
+    )
+  } else {
+    NULL
+  }
+
+  return(list(means = means, meanLines = meanLines))
+}  # End .rainMeansAndLines()
+
+
+
+# .rainCaption() ----
+# CSS formatting works through ggtext::element_markdown(); see .rainFillPlot()
+.rainCaption <- function(options, dataInfo, intervalBounds, warningAxisLimits, errorVioSides) {
+
+  exclusions <- if (dataInfo$numberOfExclusions > 0) {
+    gettextf("Not shown are %s observations due to missing data.", dataInfo$numberOfExclusions)
+  } else {
+    NULL
+  }
+
+  if (options$meanIntervalCustom) {
+    if (intervalBounds$successfulComputation) {
+      meanInterval <- gettextf("Interval around mean is custom.")
+    } else {
+      meanInterval <- gettextf("<span style = 'color: darkorange'>Error with custom interval: Specified number of clouds does not match clouds in plot.</span>")
+    }
+  } else if (options$meanInterval) {
+    if (options$meanIntervalOption == "sd") {
+      meanInterval <- gettextf("Interval around mean represents ± 1 standard deviation.")
+    } else if (options$observationId == "" && options$meanIntervalOption == "ci" && options$meanCiAssumption) {
+      meanInterval <- paste0(
+        "Interval around mean represents ",
+        options$meanCiWidth * 100,
+        "% confidence interval;<br>confidence intervals were computed independently for each group.")
+    } else {
+      meanInterval <- NULL
+    }
+  } else {
+    meanInterval <- NULL
+  }
+
+  jitter <- if (options$jitter) {
+    gettextf("Points are slightly jittered from their true values.")
+  } else {
+    NULL
+  }
```

Comment:
```suggestion
  jitter <- if (options$jitter) {
    gettextf("Points are slightly jittered from their true values.")
  }
```
because `a <- if (FALSE) 1` implies that `a` is `NULL`.

Reply:
I still see this issue on some other places

Reply:
Dropped it where I found it. All good?